lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -151,38 +347,58 @@ try
 
   $ps = [System.Management.Automation.PowerShell]::Create()
   $ps.Runspace = $runspace
-
   [Void]$ps.AddScript($ourFunctions)
   $ps.Invoke()
 
+
   if(!$environmentVariables){
+    $ps = [System.Management.Automation.PowerShell]::Create()
+    $ps.Runspace = $runspace
     $environmentVariables = $ps.AddCommand('Get-ProcessEnvironmentVariables').Invoke()
   }
 
   if($PSVersionTable.PSVersion -le [Version]'2.0'){
     if(!$psVariables){
       $psVariables = $ps.AddScript('Get-Variable').Invoke()
     }
+
+    $ps = [System.Management.Automation.PowerShell]::Create()
+    $ps.Runspace = $runspace
     [void]$ps.AddScript('Get-Variable -Scope Global | Remove-Variable -Force -ErrorAction SilentlyContinue -WarningAction SilentlyContinue')
     $ps.Invoke()
+
+    $ps = [System.Management.Automation.PowerShell]::Create()
+    $ps.Runspace = $runspace
     [void]$ps.AddCommand('Reset-ProcessPowerShellVariables').AddParameter('psVariables', $psVariables)
     $ps.Invoke()
   }
 
+  $ps = [System.Management.Automation.PowerShell]::Create()
+  $ps.Runspace = $runspace
   [Void]$ps.AddCommand('Reset-ProcessEnvironmentVariables').AddParameter('processVars', $environmentVariables)
   $ps.Invoke()
 
+  $ps = [System.Management.Automation.PowerShell]::Create()
+  $ps.Runspace = $runspace
```

Comment:
Would you mind adding some comments as to that in the code to reduce the confusion for when we've forgotten this conversation later?